Tests/BugsnagSessionTrackerTest.m

```diff
@@ -25,73 +25,100 @@ - (void)setUp {
+    XCTAssertNil(session.user);
+    XCTAssertFalse(session.autoCaptured);
+}
+
+- (void)testStartNewSessionWithUser {
+    [self.configuration setUser:@"123" withName:@"Bill" andEmail:nil];
+    XCTAssertNil(self.sessionTracker.currentSession);
+    [self.sessionTracker startNewSession];
+    BugsnagSession *session = self.sessionTracker.currentSession;
+
+    XCTAssertNotNil(session);
+    XCTAssertNotNil(session.sessionId);
+    XCTAssertTrue([[NSDate date] timeIntervalSinceDate:session.startedAt] < 1);
+    XCTAssertEqual(session.user.name, @"Bill");
+    XCTAssertEqual(session.user.userId, @"123");
+    XCTAssertNil(session.user.emailAddress);
+    XCTAssertFalse(session.autoCaptured);
+}
+
+- (void)testStartNewAutoCapturedSession {
+    XCTAssertNil(self.sessionTracker.currentSession);
+    [self.sessionTracker startNewSessionIfAutoCaptureEnabled];
+    BugsnagSession *session = self.sessionTracker.currentSession;
+
+    XCTAssertNotNil(session);
+    XCTAssertNotNil(session.sessionId);
+    XCTAssertTrue([[NSDate date] timeIntervalSinceDate:session.startedAt] < 1);
+    XCTAssertNil(session.user.name);
+    XCTAssertNil(session.user.userId);
+    XCTAssertNil(session.user.emailAddress);
+    XCTAssertTrue(session.autoCaptured);
+}
+
+- (void)testStartNewAutoCapturedSessionWithUser {
+    [self.configuration setUser:@"123" withName:@"Bill" andEmail:@"bill@example.com"];
+    XCTAssertNil(self.sessionTracker.currentSession);
+    [self.sessionTracker startNewSessionIfAutoCaptureEnabled];
+    BugsnagSession *session = self.sessionTracker.currentSession;
+
+    XCTAssertNotNil(session);
+    XCTAssertNotNil(session.sessionId);
+    XCTAssertTrue([[NSDate date] timeIntervalSinceDate:session.startedAt] < 1);
+    XCTAssertEqual(session.user.name, @"Bill");
+    XCTAssertEqual(session.user.userId, @"123");
+    XCTAssertEqual(session.user.emailAddress, @"bill@example.com");
+    XCTAssertTrue(session.autoCaptured);
+}
+
+- (void)testStartNewAutoCapturedSessionWithAutoCaptureDisabled {
+    XCTAssertNil(self.sessionTracker.currentSession);
+    self.configuration.shouldAutoCaptureSessions = NO;
+    [self.sessionTracker startNewSessionIfAutoCaptureEnabled];
+    BugsnagSession *session = self.sessionTracker.currentSession;
+
+    XCTAssertNil(session);
 }
 
 - (void)testUniqueSessionIds {
-    [self.sessionTracker startNewSession:[NSDate date]
-                                withUser:self.user
-                            autoCaptured:NO];
+    [self.sessionTracker startNewSession];
     BugsnagSession *firstSession = self.sessionTracker.currentSession;
 
-    [self.sessionTracker startNewSession:[NSDate date]
-                                withUser:self.user
-                            autoCaptured:NO];
+    [self.sessionTracker startNewSession];
 
     BugsnagSession *secondSession = self.sessionTracker.currentSession;
     XCTAssertNotEqualObjects(firstSession.sessionId, secondSession.sessionId);
 }
 
 - (void)testIncrementCounts {
```

Comment:
Is it possible to test unhandled counts? Or is that already covered elsewhere?

Reply:
It needs an integration test case. I’ll note to add one.